framework/src/test/java/org/tron/core/actuator/utils/ProposalUtilTest.java

```diff
@@ -0,0 +1,302 @@
+import org.tron.common.application.TronApplicationContext;
+import org.tron.common.utils.FileUtil;
+import org.tron.common.utils.ForkUtils;
+import org.tron.core.Constant;
+import org.tron.core.config.DefaultConfig;
+import org.tron.core.config.args.Args;
+import org.tron.core.db.Manager;
+import org.tron.core.exception.ContractValidateException;
+import org.tron.core.store.DynamicPropertiesStore;
+import org.tron.core.utils.ProposalUtil;
+import org.tron.core.utils.ProposalUtil.ProposalType;
+
+@Slf4j(topic = "actuator")
+public class ProposalUtilTest {
+  private static final String dbPath = "output_ProposalUtil_test";
+  private static final long LONG_VALUE = 100_000_000_000_000_000L;
+  public static Application AppT;
+  private static TronApplicationContext context;
+  private static Manager dbManager;
+
+  /**
+   * Init .
+   */
+  @BeforeClass
+  public static void init() {
+    Args.setParam(new String[] {"--output-directory", dbPath}, Constant.TEST_CONF);
+    context = new TronApplicationContext(DefaultConfig.class);
+    dbManager = context.getBean(Manager.class);
+    AppT = ApplicationFactory.create(context);
+  }
+
+  /**
+   * Release resources.
+   */
+  @AfterClass
+  public static void destroy() {
+    Args.clearParam();
+    AppT.shutdownServices();
+    AppT.shutdown();
+
+    context.destroy();
+    if (FileUtil.deleteDir(new File(dbPath))) {
+      logger.info("Release resources successful.");
+    } else {
+      logger.info("Release resources failure.");
+    }
+  }
+
+  @Test
+  public void validProposalTypeCheck() {
+
+    Assert.assertEquals(false, ProposalType.contain(40));
+    Assert.assertEquals(false, ProposalType.contain(-1));
+    Assert.assertEquals(true, ProposalType.contain(2));
+
+
+    Assert.assertEquals(null, ProposalType.getEnumOrNull(-2));
+    Assert.assertEquals(ProposalType.ALLOW_TVM_SOLIDITY_059, ProposalType.getEnumOrNull(32));
+
+    long code = -1;
+    try {
+      ProposalType.getEnum(code);
+      Assert.assertTrue(false);
+    } catch (ContractValidateException e) {
+      Assert.assertTrue(e instanceof ContractValidateException);
+      Assert.assertEquals("Does not support code : " + code, e.getMessage());
+    }
+
+    code = 32;
+    try {
+      Assert.assertEquals(ProposalType.ALLOW_TVM_SOLIDITY_059, ProposalType.getEnum(code));
+    } catch (ContractValidateException e) {
+      Assert.assertFalse(e instanceof ContractValidateException);
+    }
+  }
+
+  @Test
+  public void validateCheck() {
+    ProposalUtil actuatorUtil = new ProposalUtil();
+    DynamicPropertiesStore dynamicPropertiesStore = null;
+    ForkUtils forkUtils = null;
+    long invalidValue = -1;
+
+    try {
+      actuatorUtil.validator(dynamicPropertiesStore, forkUtils,
+          ProposalType.ACCOUNT_UPGRADE_COST.getCode(), invalidValue);
+      Assert.assertTrue(false);
+    } catch (ContractValidateException e) {
+      Assert.assertTrue(e instanceof ContractValidateException);
```

Comment:
always true because of catching this type of exception, Asserting LONG_VALUE_ERROR is better, I think

Reply:
Yeah. Will fix it soon.